src/components/schema/schema_item.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { CommonProps } from '../common';
+import classNames from 'classnames';
+import { OuiButtonIcon, OuiButtonIconPropsForButton } from '../button';
+import { OuiToolTip, OuiToolTipProps } from '../tool_tip';
+import { IconType } from '../icon';
+import { OuiToken } from '../token';
+
+export interface Action extends OuiButtonIconPropsForButton {
+  tooltip?: Omit<OuiToolTipProps, 'children'>;
+}
+
+/**
+ * Props for the OuiSchemaItem component.
+ * @public
+ */
+export type OuiSchemaItemProps = CommonProps & {
+  /** The icon type to display. */
+  iconType?: IconType;
+  /** The label to display. */
+  label: string;
+  /** An array of actions to display. Limit 2 */
+  actions?: Action[];
+  /** Whether the item is compressed. */
+  compressed?: boolean;
+  /** Whether the item is displayed with a panel. */
+  withPanel?: boolean;
```

Comment:
Is there precedence for using `with`? I feel like a more common naming convention would be `hasPanel`

Reply:
`droppable` uses `withPanel`

`contextMenu` uses `hasPanel`

`pageBody` uses `panelled`

So I don't think we have a consistent pattern today. Seeing that the current implementation is pure style (and doesn't actually use existing panel components or classes), I think I'd maybe choose `withPanelStyling` or something like that.

Reply:
I'd personally avoid `withPanelStyling` since its unnecessarily long, but im okay with `withPanel`, `hasPanel` or even `panelled`. Whichever you want to align on. From a user's perspective, as long as it does what the docs say, i dont care as much what its called :)